File: src/services/tree-sitter/__tests__/fixtures/java/sample.java
package com.example;

/**
 * JavaClass is a sample class.
 */
public class JavaClass {
    private String name;

    public JavaClass(String name) {
        this.name = name;
    }

    /**
     * getName returns the name.
     */
    public String getName() {
        return this.name;
    }

    public int calculate(int a, int b) {
        return a + b;
    }

    public static class InnerClass {
        public String innerMethod() {
            return "inner";
        }
    }
}

class Main {
    public static void main(String[] args) {
        JavaClass obj = new JavaClass("example");
        System.out.println(obj.getName());
        System.out.println(obj.calculate(5, 10));
    }

    public static int topLevelFunc(int x) {
        JavaClass obj = new JavaClass("top-level");
        return obj.calculate(x, 10);
    }
}

File: src/services/tree-sitter/__tests__/fixtures/kotlin/sample.kt
package com.example

/**
 * KotlinClass is a sample class.
 */
class KotlinClass(var name: String) {

    /**
     * getName returns the name.
     */
    fun getName(): String {
        return this.name
    }

    fun calculate(a: Int, b: Int): Int {
        return a + b
    }

    companion object {
        fun companionMethod(): String {
            return "companion"
        }
    }
}

/**
 * topLevelFunc is a top-level function.
 */
fun topLevelFunc(x: Int): Int {
    val obj = KotlinClass("example")
    println(obj.getName())
    return obj.calculate(x, 10)
}

fun main() {
    topLevelFunc(5)
}

data class DataClass(val id: Int)

object Singleton {
    fun singletonMethod() {
        println("singleton")
    }
}

